docs/src/tutorials/getting_started/design_patterns_for_larger_models.jl

```diff
@@ -0,0 +1,645 @@
+ * `optimizer` : an object that can be passed to `JuMP.Model` to construct a new
+   JuMP model.
+ * `data_filename` : the filename of a JSON file containing the data for the
+   problem.
+ * `config` : an object to control the type of knapsack model constructed.
+   Valid options are:
+    * `BinaryKnapsack()`
+    * `IntegerKnapsack()`
+
+## Returns
+
+ * If an optimal solution exists: a `JuMP.DenseAxisArray` that maps the `String`
+   name of each object to the number of objects to pack into the knapsack.
+ * Otherwise, `nothing`, indicating that the problem does not have an optimal
+   solution.
+
+## Examples
+
+```julia
+solution = solve_knapsack(GLPK.Optimizer, "path/to/data.json", BinaryKnapsack())
+```
+
+```julia
+solution = solve_knapsack(
+    MOI.OptimizerWithAttributes(GLPK.Optimizer, "msg_lev" => 0),
+    "path/to/data.json",
+    IntegerKnapsack(),
+)
+```
+"""
+function solve_knapsack(
+    optimizer,
+    data_filename::String,
+    config::_AbstractConfiguration,
+)
+    return _solve_knapsack(optimizer, _read_data(data_filename), config)
+end
+
+end
+
+# Finally, you can call your model:
+
+import .KnapsackModel
+
+KnapsackModel.solve_knapsack(
+    GLPK.Optimizer,
+    joinpath(@__DIR__, "data", "knapsack.json"),
+    KnapsackModel.BinaryKnapsack(),
+)
+
+# !!! note
+#     The `.` in `.KnapsackModel` denotes that it is a submodule and not a
+#     separate package that we installed with `Pkg.add`. If you put the
+#     `KnapsackModel` in a separate file, load it with:
+#     ```julia
+#     include("path/to/KnapsackModel.jl")
+#     import .KnapsackModel
+#     ```
+
+# ## Add tests
+
+# As a final step, you should add tests for your model. This often means testing
+# on a small problem for which you can work out the optimal solution by hand.
+# The Julia standard library `Test` has good unit-testing functionality.
+
+import .KnapsackModel
+using Test
+
+@testset "KnapsackModel" begin
+    @testset "feasible_binary_knapsack" begin
+        x = KnapsackModel.solve_knapsack(
+            GLPK.Optimizer,
+            joinpath(@__DIR__, "data", "knapsack.json"),
+            KnapsackModel.BinaryKnapsack(),
+        )
+        @test isapprox(x["apple"], 1, atol = 1e-5)
+        @test isapprox(x["banana"], 0, atol = 1e-5)
+        @test isapprox(x["cherry"], 0, atol = 1e-5)
+        @test isapprox(x["date"], 1, atol = 1e-5)
+        @test isapprox(x["eggplant"], 1, atol = 1e-5)
+    end
+    @testset "feasible_integer_knapsack" begin
+        x = KnapsackModel.solve_knapsack(
+            GLPK.Optimizer,
+            joinpath(@__DIR__, "data", "knapsack.json"),
+            KnapsackModel.IntegerKnapsack(),
+        )
+        @test isapprox(x["apple"], 0, atol = 1e-5)
+        @test isapprox(x["banana"], 0, atol = 1e-5)
+        @test isapprox(x["cherry"], 0, atol = 1e-5)
+        @test isapprox(x["date"], 5, atol = 1e-5)
+        @test isapprox(x["eggplant"], 0, atol = 1e-5)
+    end
+    @testset "feasible_binary_knapsack" begin
```

Comment:
Should be `"infeasible_binary_knapsack"`?

Reply:
yes I believe so